swap/swap.go

```diff
@@ -419,6 +421,103 @@ func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	return balances, nil
 }
 
+// Cheques returns all known last sent and received cheques, grouped by peer
+func (s *Swap) Cheques() (map[enode.ID]map[string]*Cheque, error) {
```

Comment:
Based solely on the function description, this function could just be implemented by calling `SentCheques` and `ReceivedCheques`. Why did you choose not to re-use those functions, but implement it fully by yourself?

Reply:
it's certainly an option, but also consider that:
- we would have to go through the store twice, one for each call of the functions.
  - same applies for going through the SWAP peers in memory, although it's not as costly.
- we would have to add an additional function that merges the results of each of the functions, which are structurally different maps from the one that `Cheques` returns.

it's an option though; feedback from others is welcome.

Reply:
@ralph-pichler i see your point, and i agree.

however, before i change this, i want to ask what your opinion is on removing the `SentCheque(s)` and `ReceivedCheque(s)` functions.

do you see a use case for querying only the received or sent cheques? (be it per peer or for all peers)

Reply:
They are probably not that useful. Even if we only need one of them we can just do the `Cheques` call and ignore the rest. I'm ok with removing them.

Reply:
addressed.